posts/2025-07-01-25.0.0.7-beta.adoc

```diff
@@ -0,0 +1,301 @@
+// important because it is supposed to grab the readers attention.  Make sure to keep the blank lines
+//
+// Throughout the doc, replace 25.0.0.7-beta with the version number of Open Liberty, eg: 22.0.0.2-beta
+// // // // // // // //
+
+The link:{url-about}[Open Liberty] 25.0.0.7-beta includes the following beta features (along with link:{url-prefix}/docs/latest/reference/feature/feature-overview.html[all GA features]):
+
+* <<SUB_TAG_0, MicroProfile OpenAPI 4.1>>
+* <<SUB_TAG_1, MicroProfile 7.1>>
+* <<SUB_TAG_2, MicroProfile Telemetry 2.1>>
+
+// // // // // // // //
+// In the preceding section:
+// Change SUB_FEATURE_TITLE to the feature that is included in this release and
+// change the SUB_TAG_1/2/3 to the heading tags
+//
+// However if there's only 1 new feature, delete the previous section and change it to the following sentence:
+// "The link:{url-about}[Open Liberty] 25.0.0.7-beta includes SUB_FEATURE_TITLE"
+// // // // // // // //
+
+See also link:{url-prefix}/blog/?search=beta&key=tag[previous Open Liberty beta blog posts].
+
+// // // // DO NOT MODIFY THIS COMMENT BLOCK <GHA-BLOG-TOPIC> // // // // 
+// Blog issue: https://github.com/OpenLiberty/open-liberty/issues/31811
+// Contact/Reviewer: Azquelt
+// // // // // // // // 
+[#SUB_TAG_0]
+== MicroProfile OpenAPI 4.1
+
+
+OpenAPI is a standardised way of documenting REST APIs in a JSON or YAML format. MicroProfile OpenAPI helps you generate and serve OpenAPI documentation for your REST applications built using JAX-RS or Jakarta restfulWS. This can be useful for developers to test out the API during development, or for people using the API in production.
+
+New in this beta is MicroProfile OpenAPI 4.1. To use it, first add the feature to your server.xml:
+
+```xml
+<featureManager>
+    <feature>mpOpenAPI-4.1</feature>
+</featureManager
+```
+
+Then deploy your rest application and then visit `/openapi` on your liberty server to view the OpenAPI documentation, or visit `/openapi/ui` for a user interface that lets you browse through it. You can enhance the documentation using link:https://download.eclipse.org/microprofile/microprofile-open-api-4.1/apidocs/[annotations] in your code to add things like textual descriptions to each method. The annotations are available as a maven dependency:
+
+```xml
+<dependency>
+    <groupId>org.eclipse.microprofile.openapi</groupId>
+    <artifactId>microprofile-openapi-api</artifactId>
+    <version>4.1</version>
+    <scope>provided</scope>
+</dependency>
+```
+
+MicroProfile OpenAPI 4.1 is a very small release, which adds:
+- The ability to set the link:https://spec.openapis.org/oas/v3.1.0.html#fixed-fields[`jsonSchemaDialect` field on the OpenAPI object] using the link:https://download.eclipse.org/microprofile/microprofile-open-api-4.1/apidocs/org.eclipse.microprofile.openapi/org/eclipse/microprofile/openapi/models/OpenAPI.html#setJsonSchemaDialect(java.lang.String[model API])
```

Comment:
```suggestion
- The ability to set the link:https://spec.openapis.org/oas/v3.1.0.html#fixed-fields[`jsonSchemaDialect` field on the OpenAPI object] using the link:https://download.eclipse.org/microprofile/microprofile-open-api-4.1/apidocs/org.eclipse.microprofile.openapi/org/eclipse/microprofile/openapi/models/OpenAPI.html#setJsonSchemaDialect(java.lang.String)[model API]
```

I'm fairly sure the link here is broken, and I think it should be this, but I wouldn't be certain without checking the output.